### PR TITLE
Support runv exec

### PR DESCRIPTION
**Pull request by gao-feng**
`runv -id 'xx' start ` will create runv container whose id is 'xx'
`runv -id 'xx' exec process.json` will execute program in container 'xx'

### CODE REVIEW

File: runv.go [hunk 1]
```diff
@@ -82,6 +86,73 @@ func getTtySize(outFd uintptr, isTerminalOut bool) (int, int) {
...
+
+	pwd, err := filepath.Abs(".")
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return nil, err
+	}
+
+	state := specs.State{
+		Version: specs.Version,
+		ID:      vmId,
```
Comment (laijs) [inline runv.go]: the id should be the containerid which is repsented by podid in this context.

File: runv.go [hunk 2]
```diff
@@ -82,6 +86,73 @@ func getTtySize(outFd uintptr, isTerminalOut bool) (int, int) {
...
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return nil, err
+	}
+
+	defer func() {
+		if err != nil {
+			os.RemoveAll(podPath)
+		}
+	}()
```
Comment (laijs) [inline runv.go]: here is the end of the function, it is awkward to a defer

File: hypervisor/init_comm.go [hunk 3]
```diff
@@ -11,8 +11,8 @@ import (
 
 // Message
 type DecodedMessage struct {
-	code    uint32
-	message []byte
+	Code    uint32
+	Message []byte
 }
```
Comment (laijs) [inline hypervisor/init_comm.go]: use a separate commit to do the rename.

File: exec.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+			}
+		}
+		// Discard errors due to pipe interruption
+		sendStdin <- nil
+	}()
+
+	if err := <-receiveStdout; err != nil {
+		return -1, err
+	}
+	sendStdin <- nil
```
Comment (laijs) [inline exec.go]: I can't understand this line

File: runv.go (hunk 2 above)
Comment (gao-feng) [inline runv.go]: I whould like to store more information in state.json, the podid can be got by directory name. I also consider to store containerid not vmid in state.json. Let's decide it later. BTW, the state.json of runc is not compatible with ocf right now.

File: hypervisor/init_comm.go (hunk 3 above)
Comment (gao-feng) [inline hypervisor/init_comm.go]: There is no big difference to seperate one commit to two,
exec support depends on this change, so they are in the same commit.
there is nothing wrong.

File: exec.go (hunk 4 above)
Comment (gao-feng) [inline exec.go]: Me too, just copy it from hyper/client/hijack.go#hijack, with my test, this line is needed

File: runv.go (hunk 1 above)
Comment (gao-feng) [inline runv.go]: I whould like to store more information in state.json, the podid can be got by directory name. I also consider to store containerid not vmid in state.json. Let's decide it later. BTW, the state.json of runc is not compatible with ocf right now.

**Comment by laijs** [comment]
LGTM

**Pull request merged**